web/src/components/NavigationBarScrollContainer.tsx

```diff
@@ -51,16 +52,34 @@ const ScrollContainer = styled.div<{ showArrowContainer: boolean }>`
   }
 `
 
-const getScrollableWidth = (scrollContainer: RefObject<HTMLDivElement>): number => {
+const getScrollableWidth = <T extends HTMLSpanElement | HTMLDivElement | HTMLParagraphElement>(
+  scrollContainer: RefObject<T>
+): number => {
   if (scrollContainer.current) {
     const { scrollWidth, clientWidth } = scrollContainer.current
     return scrollWidth - clientWidth
   }
   return 0
 }
+const getInitialScrollPosition = (activeIndex: number): number => {
+  const navigationBar = document.getElementById('navigation-bar')
+  if (!navigationBar) {
+    return 0
+  }
+  const navBarOffset = navigationBar.offsetLeft
+  const elementOffset = navigationBar.getElementsByTagName('div')[activeIndex]?.offsetLeft
+  return elementOffset ? navBarOffset + elementOffset : 0
+}
+
+const NavigationBarScrollContainer = ({ children, direction, activeIndex }: PropsType): ReactElement => {
+  const [scrollPosition, setScrollPosition] = useState<number>(0)
+  const { toggle, refCallback, ref: scrollContainerRef } = useRefWithCallback()
 
-const NavigationBarScrollContainer = ({ children, direction, scrollContainerRef }: PropsType): ReactElement => {
-  const [scrollPosition, setScrollPosition] = useState(0)
+  useEffect(() => {
+    if (activeIndex > 0 && scrollContainerRef.current) {
+      scrollContainerRef.current.scrollTo({ left: getInitialScrollPosition(activeIndex) })
+    }
+  }, [activeIndex, scrollContainerRef, toggle])
```

Comment:
this is used instead of unsafe `ref.current` check`useRefWithCallback` hook

Reply:
Nice idea. I think this could be improved a little further though, there is no need for the toggle or the useEffect, instead we can just pass the callback to the hook and execute it directly there. Furthermore, I did some minor adjustments and renamed the hook to `useCallbackRef` in accordance with the react docs.
See #780. Feel free to reject it completely or to just apply some changes.